scilpy/io/utils.py

```diff
@@ -247,36 +247,87 @@ def add_bbox_arg(parser):
         Parser.
     mandatory: bool
         Whether this argument is mandatory.
+    input_output: bool
+        Whether this argument should expect both input and output bases or not.
+        If set, the sh_basis argument will expect first the input basis,
+        followed by the output basis.
     """
-    choices = ['descoteaux07', 'tournier07']
-    def_val = 'descoteaux07'
+    if input_output:
+        nargs = 2
+        def_val = ['descoteaux07_legacy', 'tournier07']
+        input_output_msg = '\nBoth the input and output bases are ' +\
+                           'required, in that order.'
+    else:
+        nargs = 1
+        def_val = ['descoteaux07_legacy']
+        input_output_msg = ''
+
+    choices = ['descoteaux07', 'tournier07', 'descoteaux07_legacy',
+               'tournier07_legacy']
     help_msg = 'Spherical harmonics basis used for the SH coefficients. ' +\
-               '\nMust be either \'descoteaux07\' or \'tournier07\'' +\
+               input_output_msg +\
+               '\nMust be either \'descoteaux07\', \'tournier07\', \n' +\
+               '\'descoteaux07_legacy\' or \'tournier07_legacy\'' +\
                ' [%(default)s]:\n' +\
-               '    \'descoteaux07\': SH basis from the Descoteaux et al.\n' +\
-               '                      MRM 2007 paper\n' +\
-               '    \'tournier07\'  : SH basis from the Tournier et al.\n' +\
-               '                      NeuroImage 2007 paper.'
+               '    \'descoteaux07\'       : SH basis from the Descoteaux et al.\n' +\
+               '                             MRM 2007 paper\n' +\
+               '    \'tournier07\'         : SH basis from the new Tournier et al.\n' +\
+               '                             NeuroImage 2019 paper, as in MRtrix 3.\n' +\
+               '    \'descoteaux07_legacy\': SH basis from the legacy Dipy\n' +\
+               '                             implementation of the Descoteaux et al.\n' +\
+               '                             MRM 2007 paper\n' +\
+               '    \'tournier07_legacy\'  : SH basis from the legacy Tournier et al.\n' +\
+               '                             NeuroImage 2007 paper.'
 
     if mandatory:
         arg_name = 'sh_basis'
     else:
         arg_name = '--sh_basis'
 
-    parser.add_argument(arg_name,
+    parser.add_argument(arg_name, nargs=nargs,
                         choices=choices, default=def_val,
                         help=help_msg)
 
 
+def interpret_sh_basis(args):
+    """
+    Interpret the input from args.sh_basis. If two SH bases are given,
+    both input/output sh_basis and is_legacy are returned.
+
+    Parameters
+    ----------
+    args : argparser
+        Argparser from a script.
+
+    Returns
+    -------
+    sh_basis : string
+        Spherical harmonic basis name.
+    is_legacy : bool
+        Whether or not the SH basis is in its legacy form.
+    """
+    if len(args.sh_basis) == 2:
```

Comment:
it works but you could also simply do
```
 basis = 'descoteaux07' if 'descoteaux07' in sh_basis_name else 'tournier07'
 legacy = 'legacy' in sh_basis_name
```

Reply:
Yes but I prefer the more "general" way, without assuming descoteaux07 or tournier07. @arnaudbore ? I think Charles refers to lines 313 to 316 and 319 to 320.

Reply:
Is it better now?

Reply:
cool